gslab_scons/_release_tools.py

```diff
@@ -223,29 +220,42 @@ def up_to_date(mode = 'scons', directory = '.', scons_local_path = None):
         # argument is actually a SCons directory.
         # We use the fact that running scons outside of SCons directory
         # produces the message: "No SConstruct file found."
-        if [True for out in output if re.search('No SConstruct file found', out)]:
-            raise ReleaseError('up_to_date(mode = scons) must be run on a '
+        if check_list_for_regex('No SConstruct file found', output):
+            raise ReleaseError('up_to_date(mode = scons) must be run on an '
                                'SCons directory.')
+        # Next determine if SCons could start to run.
+        # We use the fact that our SCons notifies when it starts the build process.
+        elif not check_list_for_regex('scons: Reading SConscript files', output):
+            raise ReleaseError('up_to_date(mode = scons) must be able to begin '
+                               'SCons build process.')
+
```

Comment:
I'm still getting this error when I run outside the SCons directory with no SConstruct

```bash
Traceback (most recent call last):
  File "//anaconda/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "//anaconda/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/anaconda/lib/python2.7/site-packages/GSLab_Tools-4.1.0-py2.7.egg/gslab_scons/release.py", line 129, in <module>
    main()
  File "/anaconda/lib/python2.7/site-packages/GSLab_Tools-4.1.0-py2.7.egg/gslab_scons/release.py", line 28, in main
    inspect_repo(scons_local_path = scons_local_path)
  File "/anaconda/lib/python2.7/site-packages/GSLab_Tools-4.1.0-py2.7.egg/gslab_scons/release.py", line 117, in inspect_repo
    if not _release_tools.up_to_date(mode = 'scons', scons_local_path = scons_local_path):
  File "//anaconda/lib/python2.7/site-packages/GSLab_Tools-4.1.0-py2.7.egg/gslab_scons/_release_tools.py", line 229, in up_to_date
    raise ReleaseError('up_to_date(mode = scons) must be able to begin '
gslab_scons._exception_classes.ReleaseError: up_to_date(mode = scons) must be able to begin SCons build process.
```

Reply:
Clearly the l229 ReleaseError works properly but not the l224 one

Reply:
```
template (4.1.0)$ scons

scons: *** No SConstruct file found.
File "/anaconda/lib/python2.7/site-packages/scons-2.5.1/SCons/Script/Main.py", line 912, in _main
```

Reply:
That's because the error you get when you run in default mode is actually 

```
python: can't open file './run.py': [Errno 2] No such file or directory
```

which makes sense

Reply:
Yup.

```
template (4.1.0)$ python -m gslab_scons.release version=test_sep6 readme=../README.md scons_local_path=None
Traceback (most recent call last):
  File "//anaconda/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "//anaconda/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/anaconda/lib/python2.7/site-packages/GSLab_Tools-4.1.0-py2.7.egg/gslab_scons/release.py", line 129, in <module>
    main()
  File "/anaconda/lib/python2.7/site-packages/GSLab_Tools-4.1.0-py2.7.egg/gslab_scons/release.py", line 28, in main
    inspect_repo(scons_local_path = scons_local_path)
  File "/anaconda/lib/python2.7/site-packages/GSLab_Tools-4.1.0-py2.7.egg/gslab_scons/release.py", line 117, in inspect_repo
    if not _release_tools.up_to_date(mode = 'scons', scons_local_path = scons_local_path):
  File "//anaconda/lib/python2.7/site-packages/GSLab_Tools-4.1.0-py2.7.egg/gslab_scons/_release_tools.py", line 224, in up_to_date
    raise ReleaseError('up_to_date(mode = scons) must be run on an '
gslab_scons._exception_classes.ReleaseError: up_to_date(mode = scons) must be run on an SCons directory.
```

I guess adding an `or` and allow the second pattern would fix this.